src/NServiceBus.Core/MessageSession.cs

```diff
@@ -36,9 +36,9 @@ public Task Subscribe(Type eventType, SubscribeOptions options)
             return messageOperations.Subscribe(context, eventType, options);
         }
 
-        public Task Unsubscribe(Type eventType, UnsubscribeOptions options)
+        public Task Unsubscribe(Type eventType, UnsubscribeOptions unsubscribeOptions)
         {
-            return messageOperations.Unsubscribe(context, eventType, options);
+            return messageOperations.Unsubscribe(context, eventType, unsubscribeOptions);
```

Comment:
Since we added Guards to RunningEndpointInstance we should add them here as well?

Reply:
Not sure, usually you try to add these to the public surface, will this object be accessible by user code? It doesn't hurt to have but for internal code you usually would use asserts as that would indicate a state that it should have never reached in the first place.

Reply:
Since IMessageSession is what we provide to the users via DI these methods are the ones that will be used outside handlers for most users? (if we don't do it here we shouldn't add them on RunningEndpointInstance either IMO)

Reply:
Also note that RunningEndpoint instance will call these methods so adding them here will kill 2 birds with one stone?

Reply:
@andreasohlund I've added a commit to add the guards to MessageSession but not sure it is needed. When we supply a user an `IMessageSession`, which object(s) can this be? I thought that could only be `RunningEndpointInstance`? Meaning, `MessageSession` references are never directly accessible to the user?

Reply:
We pass it to the running endpoint instance and all the features

https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/StartableEndpoint.cs#L57
https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/StartableEndpoint.cs#L59

so arguably you could remove the guards from this class now?